### PR TITLE
RDM-6719: Add Banner Feature to CCD UI

### PR DESCRIPTION
**Pull request by kiran-yenigala-hmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDM-6719

### Change description ###

Add Banner to CCD UI based on each Jurisdiction configuration

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: application/src/test/resources/GetCaseTypesResponseForCCD_TestDefinition_V45.json [hunk 1]
```diff
@@ -13,6 +13,10 @@
     "description" : "Content for the Test Jurisdiction.",
```
Comment (mario-paniccia) [inline application/src/test/resources/GetCaseTypesResponseForCCD_TestDefinition_V45.json]: would it be possible to properly model the banner in Object Oriented terms with a Banner object?

File: excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/service/ImportServiceImpl.java [hunk 2]
```diff
@@ -228,9 +228,7 @@ public IdamProperties getUserDetails() {
     }
 
     private JurisdictionEntity importJurisdiction(JurisdictionEntity jurisdiction) {
-        return jurisdictionService.get(jurisdiction.getReference()).orElseGet(() -> {
-            jurisdictionService.create(jurisdiction);
-            return jurisdiction;
-        });
+        jurisdictionService.create(jurisdiction);
+        return jurisdiction;
```
Comment (MSancaktutar) [inline excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/service/ImportServiceImpl.java]: Couldn't see why we needed to do this.

Comment (kiran-yenigala-hmcts) [reply excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/service/ImportServiceImpl.java]: If Jurisdiction already exists in the DB without banner details, and if user tries to import new definition with banner details added into the definition file, then it is not going to save banner information into db, as the previous logic thinks jurisdiction already exists.

File: application/src/test/resources/GetCaseTypesResponseForCCD_TestDefinition_V45.json (hunk 1 above)
Comment (kiran-yenigala-hmcts) [reply application/src/test/resources/GetCaseTypesResponseForCCD_TestDefinition_V45.json]: We can do that, and need to change database structure as well, need to introduce new table for Banner and reference jurisdiction id in the new table..

Comment (mario-paniccia) [reply application/src/test/resources/GetCaseTypesResponseForCCD_TestDefinition_V45.json]: @kiran-yenigala-hmcts I'd recommend we do that. That we introduce a banner table. I really don't like introducing banner info into the jur table. I would also have a separate Banner worksheet in the spreadsheet dedicated to the banner. Was having a chat with @rafalkalita and he has my same concerns

Comment (mario-paniccia) [reply application/src/test/resources/GetCaseTypesResponseForCCD_TestDefinition_V45.json]: basically I'd like to see ideally 0 changes on the Jur code

File: domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapper.java [hunk 3]
```diff
@@ -221,6 +221,8 @@ private EventComplexTypeEntityToCaseEventFieldComplexListMapper() {
     @Mapping(source = "searchAliasFieldEntity.caseType.reference", target = "caseTypeId")
```
Comment (mario-paniccia) [inline domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapper.java]: commenting here because it doesn't allow me to comment on the spreadsheet files. 
I see we have added many new test spreadsheets. In my opinion this should be avoided. The reason being that this increases that maintainability effort of our tests. Because spreadsheet change, we might add new mandatory elements in the future. And when that happens we'll have to update all the test spreadsheets. 
Do we need to add CCD_TestDefinition_V46_RDM-4324.xlsx? Can't we update an existing one?
Also, I see many spreadsheets probably added to test corner case conditions in func tests or integration tests. Do we need to use func tests or int tests for that? We should be able to test all the corner cases with unit tests. To reduce the amount of int and func test to the bare minimun.

File: application/src/test/resources/GetCaseTypesResponseForCCD_TestDefinition_V45.json [hunk 4]
```diff
@@ -13,7 +13,8 @@
...
     "live_from" : "2017-01-01T00:00:00.000+0000",
     "live_until" : null,
-    "case_types" : [ ]
+    "case_types" : [ ],
+    "banners": []
```
Comment (mario-paniccia) [inline application/src/test/resources/GetCaseTypesResponseForCCD_TestDefinition_V45.json]: In CCD all UI bits of definition are not returned as part of the case type definition. Def Store provides separate endpoints to retrieve the UI definitions under the path `/api/display`. See `DisplayApiController`. Banners is a UI definition, so I think we should add retrieval of banners as a dedicated endpoint in `DisplayApiController`.  Data Store could then retrieve the banners for a certain jur by calling this endpoint.

File: domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/banner/BannerServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+            .collect(toList());
+    }
+
+    @Override
+    public List<Banner> getAll() {
+        List<BannerEntity> bannerEntities = bannerRepository.findAll();
+        return bannerEntities.stream()
+            .map(dtoMapper::map)
+            .collect(toList());
+    }
```
Comment (MSancaktutar) [inline domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/banner/BannerServiceImpl.java]: I can't any reasonable use case for this. Could we remove this or throw an UnsupportedOperationException?

Comment (kiran-yenigala-hmcts) [reply domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/banner/BannerServiceImpl.java]: Removed this method....

File: application/src/test/resources/GetCaseTypesResponseForCCD_TestDefinition_V45.json (hunk 1 above)
Comment (kiran-yenigala-hmcts) [reply application/src/test/resources/GetCaseTypesResponseForCCD_TestDefinition_V45.json]: Done changes as requested

File: excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/service/ImportServiceImpl.java (hunk 2 above)
Comment (kiran-yenigala-hmcts) [reply excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/service/ImportServiceImpl.java]: done changes

File: domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapper.java (hunk 3 above)
Comment (kiran-yenigala-hmcts) [reply domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapper.java]: done changes
